### DISCUSSION TITLE
Ability to search within a dropdown by typing.

### DISCUSSION THREAD

**Discussion by Coppertine** [ppy/osu#17020] [Ideas] [answered]
Like above, when working with long dropdown lists (i.e. List of Players in a tournament or list of skins) it's difficult to find a spesific thing within said list.
The idea is that when focused within the dropdown list, you can press the key of the first letter for the cursor to jump to the first occurence, following with another letter with jump the cursor to the list item that starts with the sequence of characters typed and etc for all characters typed. This would be simlar to selecting a spesific file in File Explorer using just your keyboard.

Another idea is to use the Dropdown box as a search bar that is typeable that would filter the list down. Similar to the song select search functionaility.

**Comment by frenzibyte** [comment]
I think this is warranting support for a "combo box" UI control instead.

**Reply by bdach** [reply]
that's not necessarily true. even normal selection dropdowns sometimes support item preselection by typing out the title of the dropdown item to preselect. see: html `<select>` element.

**Reply by frenzibyte** [reply]
Hmm, sounds like that could work with less implementation, right.

**Comment by peppy** [comment]
I'm definitely not against this proposal. I'm not sure what the best UX for this is, but regardless of what we choose the current search should be shown somewhere (unlike windows for instance, where it is hidden and arbitrarily reset after x seconds of not typing).

**Comment by Wieku** [comment]
Personally I'm not a fan of using the combo box preview for search (IIRC Windows combo boxes sometimes do this) or jumping to the first result. IMO something like this is better UX-wise:

https://user-images.githubusercontent.com/5307179/174587229-47d72e10-8709-4202-8a47-f38e764014ed.mp4

If that looks okay I can try making PoC for this in osu(-framework).

**Reply by Coppertine** [reply]
It kinda looks clunky, but that would be depending on how others think.

**Reply by Wieku** [reply]
Search box inside the combo box is automatically focused so it acts more like an indicator than actual search box.

The problem with:

> to the list item that starts with the sequence of characters typed

Is that some skins use dashes and spaces to be positioned higher, so prefix search won't work well in that case.

**Reply by Coppertine** [reply]
The reason why I initially requested for this change is due to the method of selecting players in a bracket within the Tournament Client.
![image](https://user-images.githubusercontent.com/37494321/174620803-263422e6-9fc4-441e-969f-0a4fa5e85b8e.png)
If you have many players in total, you have to scroll in a small area to select a spesific person, as well, the area above is completely empty that could be instead typed in to search.

**Reply by Coppertine** [reply]
But I do appreciate the ideas so far

**Reply by peppy** [reply]
See https://github.com/ppy/osu/discussions/17020#discussioncomment-2270422

**Comment by Flutterish** [comment]
If you go with the search box, I think it should be in the same place the "currently selected" option is

**Comment by frenzibyte** [comment, marked as answer]
Tracking in https://github.com/ppy/osu/issues/21784

**Comment by JDECOMBE** [comment]
I feel like the ability to pin some items at the top of the list would be a great help to find often used items. 
I'm someone that uses a different skin for DT (and I'm not the only one), having the ability to quickly access the 2 or 3 skins that I use daily would be a great help.

![image](https://user-images.githubusercontent.com/25856127/236074911-b6b2b514-a3cf-4e73-ba24-1f94e0e98147.png)

If this looks good to you I can try to work on the implementation (with or without the search bar).
